### DISCUSSION TITLE
发布 @element-plus/utils 和 @element-plus/hooks 的新版本

### DISCUSSION THREAD

**Discussion by StringKe** [element-plus/element-plus#15960] [Feature Request]
Feature type: **`Other`**

## Details

- Enhanced component name: `N/A`

## Description

想使用一些包含在 utils 和 hooks 中的代码用来编写自己的组件，目前 npm 上的版本为 0.0.5 是三年的包了。

### Additional comments

_(empty)_

**Comment by kooriookami** [comment]
为什么不直接从element-plus中引用呢

**Reply by StringKe** [reply]
utils 一些工具在 element-plus 包中没找到

https://github.com/element-plus/element-plus/blob/dev/packages/element-plus/index.ts

**Reply by kooriookami** [reply]
直接引入'element-plus/es/utils'这个包;

**Reply by xxholly32** [reply]
我个人有试过，分不同的包并且发布，但是配置typescript极其复杂，而且涉及到工程化也很麻烦，utils开放会有一定的好处，但同时也会承担相应的维护成本。

**Reply by gaylen1** [reply]
无法从element-plus/es/utils导出

![image](https://github.com/element-plus/element-plus/assets/35060559/9a64749d-7d59-4b80-877c-4f66d2917aab)

导出会报这个错误

**Reply by xiaoshuoyiyi** [reply]
> 无法从element-plus/es/utils导出
> 
> ![image](https://private-user-images.githubusercontent.com/35060559/318712513-9a64749d-7d59-4b80-877c-4f66d2917aab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MTE5MDYsIm5iZiI6MTcyMTgxMTYwNiwicGF0aCI6Ii8zNTA2MDU1OS8zMTg3MTI1MTMtOWE2NDc0OWQtN2Q1OS00YjgwLTg3N2MtNGY2NmQyOTE3YWFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI0VDA5MDAwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhOTg2Y2YwMDE0Y2QzNWZlMzEzMDZmYmEzOGNiMTNhYmI1ODljNTM4YzcxNzQ0YThiOTA2ZjZmMzZlMDc1MzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CRtRbcXv24uKQwbGIiR4nwbAI-ieiaV_oRJI9jQ-7hA)
> 
> 导出会报这个错误

我也是这个错
